sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -422,7 +422,7 @@ public String getRegionName(URI locationEndpoint, com.azure.cosmos.implementatio
         }
 
         //If preferred list is not set, locationEndpoint will be default endpoint, so return the hub region
-        return this.locationInfo.availableWriteEndpointByLocation.keySet().iterator().next();
+        return this.locationInfo.availableWriteLocations.get(0);
```

Comment:
What's the difference between `availableWriteEndpointByLocation` and `availableWriteLocations`?
I am curious if `availableWriteEndpointByLocation` is getting updated with any change in the endpoints (by adding region / removing region), and if `availableWriteLocations` follows the same behavior?
This is also used in Consistency session improvements that @jeet1995 worked on, lets connect and make sure the `getRegionName()` is returning the correct region name with this change across different features.

Reply:
I think - Tomas correct me if I misread the intention - the problem is that the keySet is not an ordered list - while availableWriteLocations is.

Reply:
Fabian that is correct. Both get updated at the same time. In each update, `availableWriteEndpointByLocation` is created from the new `availableWriteLocations`. The difference is that `availableWriteEndpointByLocation` is a map so the ordering gets messed up while `availableWriteLocations` is a list.

Reply:
From the session consistency perspective - when we pass empty `preferredRegions` we are pinned to the primary region - so the first available region returned by the gateway. One weird scenario that could possibly happen is if the primary somehow changes prior to routing and once response is received both from diagnostic and session tracking perspective. cc: @tvaron3, @kushagraThapar, @FabianMeiswinkel